### ISSUE TITLE
[FR] Debugger enhancements ideas

### ISSUE THREAD

**Issue by aviatesk** [JunoLab/Juno.jl#362]
Couple of ideas to improve our debugger:

- Inline evaluation including local bindings during debugging:
    * I would be nice if selection-based inline evaluation can handle local bindings in a current scope function (which is already done in the integrated REPL)
    * would be not so difficult to implement using the same logic as https://github.com/JunoLab/Atom.jl/pull/173
- Reverse the order of workspace stack: https://github.com/JunoLab/Atom.jl/pull/175
- Watch list
     * Debugger.jl already enables this feature, but IMHO it would be much fancier if Juno has this feature as well with friendly UI.

**Comment by BioTurboNick** [comment]
If I may add: A visual indicator that the file being stepped through is not the same code the debugger is running. (Not saved or module needs to be recompiled).

**Comment by timholy** [comment]
@BioTurboNick: related, https://github.com/timholy/Revise.jl/pull/345. But that won't help with the "not saved" part.

**Comment by racinmat** [comment]
May I add idea to execute code in debug by ctrl+enter?
Now when I am at breakpoint and want to execure code, I need to copy-paste it into the REPL, becaue ctrl+enter tried to execute it in global context, even during debugging.

**Comment by aviatesk** [comment]
yeah I guess that's would be good and could easily be implemented. Will try to do when I have time.

**Comment by zhopan77** [comment]
Here's one: allow debugger to start at the top script level by clicking a button.

Naively this can be done by wrapping the whole script in a function and Juno.@enter that function. However I'm not sure how global can be dealt with, and I recall the scope rules are different at global vs. inside a function, so it's probably more complicated than that.
